TFNetworkRecLayer.py

```diff
@@ -1067,6 +1081,20 @@ def get_layer(name):
       get_layer(layer_name)
       assert layer_name in self.net.layers
 
+  def get_sub_layer(self, layer_name):
+    """
+    :param str layer_name:  The sub_layer addressed by '/' separated path.
+    :return: The sub_layer addressed in layer_name or None if no sub_layer exists
+    :rtype: LayerBase|None
+    """
+    if self.output_layers_net and layer_name in self.output_layers_net.layers:
+      return self.output_layers_net.layers[layer_name]
+    elif self.input_layers_net and layer_name in self.input_layers_net.layers:
+      return self.input_layers_net.layers[layer_name]
+    elif self.net and layer_name in self.net.layers:
+      return self.net.layers[layer_name]
```

Comment:
I'm not sure if it is a good idea to allow access to `self.net` here as well. Because this is inside the recurrent loop, thus:
* It is without the time dim. This is not consistent when you get the layer from e.g. output_layers_net or input_layers_net. But it should be consistent.
* You also cannot access the `layer.output.placeholder`, because it is inside the recurrent loop. So depending on the use case, this doesn't make sense.

I think I would throw an exception here, like:
```suggestion
      raise Exception("%r: Cannot get layer %r from outside, because it is only available inside the recurrent loop. Add 'is_output_layer':True to the layer options." % (self.parent_rec_layer, layer_name))
```